tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature

```diff
@@ -413,3 +413,25 @@ Feature: create a public link share
       | d:href       |
     Then the HTTP status code should be "207"
     And the value of the item "//d:href" in the response should match "/%base_path%\/remote.php\/dav\/public-files\/%public_token%\/file.txt$/"
+
+  @issue-6929
+  Scenario Outline: create a password-protected public link on a file with the name same to the previously deleted one
+    Given using OCS API version "<ocs_api_version>"
+    And user "Alice" has uploaded file with content "test data 1" to "/test.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | test.txt |
+      | permissions | read     |
+    And user "Alice" has deleted file "test.txt"
+    And user "Alice" has uploaded file with content "test data 2" to "/test.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | test.txt |
+      | permissions | read     |
+    When user "Alice" updates the last public link share using the sharing API with
```

Comment:
Sorry, I doubt you reproduced the bug correctly. there was a panic on the server when we tried to update the password link of a deleted file

Steps: 
- create `file1` and `link1` and delete `file1`
- create `file2` (with same name) - here we see old `link1` - web issue
- change link password `link1` -> server panic. I the test you change `link2` instead of `link1`

Reply:
Oh sorry. i missunderstood. I will change it accordingly. @ScharfViktor Thanks.